src/Service/StateHandlerInterface.php

```diff
@@ -15,71 +15,74 @@
  * limitations under the License.
  */
 
-namespace Surfnet\GsspBundle\Saml;
+namespace Surfnet\GsspBundle\Service;
 
-interface StateHandler
+use Surfnet\SamlBundle\SAML2\ReceivedAuthnRequest;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Knows and preserves the integrity of the GSSP application state.
+ */
+interface StateHandlerInterface
 {
     /**
-     * @param string $originalRequestId
-     * @return $this
+     * Stores the state of a registration request.
+     *
+     * @param Request $request
+     * @param ReceivedAuthnRequest $authnRequest
+     * @param string $stepupRequestId
      */
-    public function setRequestId($originalRequestId);
+    public function saveRegistrationRequest(Request $request, ReceivedAuthnRequest $authnRequest, $stepupRequestId);
 
     /**
-     * @return string
+     * Stores the state of a authentication request.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param \Surfnet\SamlBundle\SAML2\ReceivedAuthnRequest $authnRequest
+     * @param string $stepupRequestId
      */
-    public function getRequestId();
+    public function saveAuthenticationRequest(Request $request, ReceivedAuthnRequest $authnRequest, $stepupRequestId);
 
     /**
-     * @param string $serviceProvider
+     * @param string $message
+     *   The error message.
+     * @param string $subCode
+     *   Saml response sub code.
+     *
      * @return $this
      */
-    public function setRequestServiceProvider($serviceProvider);
-
-    /**
-     * @return string
-     */
-    public function getRequestServiceProvider();
+    public function setErrorStatus($message, $subCode);
 
     /**
-     * @param string $relayState
+     * @param $nameId
      * @return $this
      */
-    public function setRelayState($relayState);
-
-    /**
-     * @return string
-     */
-    public function getRelayState();
+    public function saveIdentityNameId($nameId);
 
     /**
-     * @param $nameId
-     * @return $this
+     * @return self
      */
-    public function saveIdentityNameId($nameId);
+    public function authenticate();
 
     /**
      * @return string
      */
-    public function getIdentityNameId();
+    public function getRequestId();
 
     /**
-     * @param string $locale
-     * @return $this
+     * @return string
      */
-    public function setPreferredLocale($locale);
+    public function getRequestServiceProvider();
 
     /**
      * @return string
      */
-    public function getPreferredLocale();
+    public function getRelayState();
 
     /**
-     * Set the current request type as registration flow.
-     *
-     * @return $this
+     * @return string
      */
-    public function setRequestTypeRegistration();
+    public function getIdentityNameId();
```

Comment:
In my opinion, get/set SubjectNameId was a better (closer to SAML spec) name for these methods.

Reply:
👍  will change it